File: ch05/elixir/bday/lib/bday.ex
defmodule Bday do
  @moduledoc """
  Documentation for `Bday`.
  """

  def run(path) do
    set =
      path
      |> File.read!()
      |> Bday.Employee.from_csv()
      |> Bday.Employee.filter_birthday(DateTime.to_date(DateTime.utc_now()))
      |> Bday.Employee.fetch()

    for employee <- set do
      employee
      |> Bday.MailTpl.full()
      |> send_email()
    end

    :ok
  end

  defp send_email({to, _, _}) do
    IO.puts("sent birthday email to #{to}")
  end
end

File: ch05/erlang/bday/src/bday.erl
-module(bday).
-export([main/1]).

main([Path]) ->
  {ok, Data} = file:read_file(Path),
  Handle = bday_employee:from_csv(binary_to_list(Data)),
  Query = bday_employee:filter_birthday(Handle, date()),  % update the date on CSV
  BdaySet = bday_employee:fetch(Query),
  Mails = [bday_mail_tpl:full(Employee) || Employee <- BdaySet],
  [send_email(To, Topic, Body) || {To, Topic, Body} <- Mails].

send_email(To, _, _) ->
  io:format("sent birthday email to ~p~n", [To]).
